File: Week 10/Java/alternatingBits.java
/**
 * Prompt:
 * 
 * Given a positive integer, check whether it has alternating bits: name, if two
 * adjacent bits will always have different values.
 * 
 * https://leetcode.com/problems/binary-number-with-alternating-bits/
 */

class Solution {
    public boolean hasAlternatingBits(int n) {
        String binary = Integer.toBinaryString(n);
        if (binary.contains("11") || binary.contains("00")) {
            return false;
        } else {
            return true;
        }
    }
}

/**
 * Same solution as my JavaScript solution, but with much better results
 * (especially regarding memory usage). Runtime of 1ms (faster than 17.63%) and
 * memory usage of 35.7mb (less than 66.37%)
 */

File: First Job Hunt/Week 10/JavaScript/alternatingBits.js
/**
 * Prompt:
 *
 * Given a positive integer, check whether it has alternating bits: namely, if two
 * adjacent bits will always have different values.
 *
 * https://leetcode.com/problems/binary-number-with-alternating-bits/
 */

/**
 * @param {number} n
 * @return {boolean}
 */

let hasAlternatingBits = (n) => {
  let binary = n.toString(2);
  console.log(binary);

  if (binary.includes("11") || binary.includes("00")) {
    return false;
  } else {
    return true;
  }
};

/**
 * I was initially over-complicating this by looping over the binary string
 * searching for any occurrence of non alternating bits before remembering
 * about the .includes method. Runtime of 88ms (faster than 28.18%)
 * and memory usage of 40.3mb (less than 5.45%)
 */
